### ISSUE TITLE
static-this - Improperly flags returned anonymous class

### ISSUE THREAD

**Issue by onehorsetown** [palantir/tslint#4579]
### Bug Report

- 5.14:
- 3..3.3333:
- CLI

#### TypeScript code being linted

```ts
interface Bar {}

class Foo {
  public static higherOrderComponent(): any {
    return class implements Bar {
      public constructor() {
        console.log(this);   // "this" is valid here but tslint is complaining
      }
    };
  }
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:all"
  ],
  "rules": {
    "static-this": true
  }
}
```

#### Actual behavior
This improperly flags this in an anonymous class returned by a static function, something that is typically used to construct higher-order components.

#### Expected behavior
`this` used in a class scoped inside a static function is valid.

**Comment by mbelsky** [comment]
I'm already looking for a solution
